### PR TITLE
[dagster-fivetran] Initial Commit

### PR DESCRIPTION
**Pull request by OwenKephart**
This a working dagster-fivetran integration. At the moment, its main functionality is to allow users to initialize a fivetran sync operation and wait until it completes.

A fivetran sync is essentially just making sure that the current state of a sink reflects all the newest data from the source since the last sync.

The resource also supports just launching the sync operation (and not polling until completion), but there's not an op that reflects this behavior at the moment, mostly because I see it as a weirder / harder to understand use case. It's pretty risky to do that, because there's no feedback mechanism to tell you if the sync operation failed or took a really long time, etc. I'm sure someone will find a use for that behavior, but they can always just create their own op that uses the resource.

I've tested this agains a real Fivetran account and it works, and the tests are created using real(ish) Fivetran response data.

Sill needs api docs.

### CODE REVIEW

File: python_modules/libraries/dagster-fivetran/dagster_fivetran/ops.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        "poll_timeout": Field(
+            Noneable(float),
+            default_value=None,
+            description="The maximum time that will waited before this operation is timed out. By "
+            "default, this will never time out.",
+        ),
+    },
+    required_resource_keys={"fivetran"},
+)
+def fivetran_sync_op(context) -> Dict[str, Any]:
```
Comment (sryza) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran/ops.py:30]: Want to start a discussion in Slack to talk through what our convention should be with respect to suffixing op names with `_op`? It seems like half our integrations do it, and half our integrations don't.

I think we generally do not encourage doing that with non-library ops - e.g. none of the ops in the hacker news jobs have the `_op` suffix.

**Comment by schrockn** [comment]
Owen what was/is your process for determining whether or not something is parameterized in the config domain versus python domain? Another way to put it is op versus op factory? Just wondering what our pattern/guidance is these days?

**Comment by OwenKephart** [comment]
> Owen what was/is your process for determining whether or not something is parameterized in the config domain versus python domain? Another way to put it is op versus op factory? Just wondering what our pattern/guidance is these days?

My general feeling is that an op factory that does nothing other than "bake-in" certain values is functionally very similar to using `.configured()`, but comes with extra (potentially unwanted) limitations.

For example, you might actually want the connector_id property of an op to change between different environments (one connector might do the s3 -> redshift sync for dev, and a different one for prod). Exposing the op factory as the only interface for setting these values makes it impossible to handle cases like this.

So the short answer is that for anything that you could conceivably want to change between environments, I prefer having configuration as an option.

There is an option of exposing an op factory in addition to the configured api, so that the somewhat unsightly:

```
my_fivetran_sync = fivetran_sync_op.configured({"connector_id": "xyz"}, name="my_fivetran_sync")
```

could be replaced with:

```
my_fivetran_sync = get_fivetran_sync_op("xyz", name="my_fivetran_sync")
```

This would just be calling .configured() under the hood. Only real problem with this setup is making things a little confusing w/ all the options.

File: python_modules/libraries/dagster-fivetran/dagster_fivetran/resources.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+        "request_retry_delay": Field(
+            float,
+            default_value=0.25,
+            description="Time (in seconds) to wait between each request retry.",
+        ),
+    },
+    description="This resource helps manage Fivetran connectors",
+)
+def fivetran_resource(context) -> FivetranResource:
+    return FivetranResource(
```
Comment (sryza) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran/resources.py:330]: It would be helpful to include docstring

**Review by sryza — COMMENTED** [review]
Discussed with Owen offline, and I'm on board with both the `_op` suffix and configurable op instead of factory. This should be good to go after adding some docstrings to the main APIs.

**Pull request merged**